spidermon/contrib/monitors/mixins/validation.py

```diff
@@ -121,11 +121,24 @@ def check_missing_required_fields_percent(
         self, field_names=None, allowed_percent=0
     ):
         if not field_names:
-            missing_percent = self.validation.errors["missing_required_field"].percent
-            self._check_missing_required_percent(missing_percent, allowed_percent)
-        else:
-            for field_name in field_names:
-                self.check_missing_required_field_percent(field_name, allowed_percent)
+            field_names = self.validation.fields
+        msgs = []
+        for field_name in field_names:
+            missing_percent = (
+                self.validation.fields[field_name].errors["missing_required_field"].percent
+            )
+            if missing_percent > allowed_percent:
+                msg = "{percent}% of required field {field} are missing!{threshold_info}".format(
```

Comment:
At least this message could be extracted in a method, since it is duplicated a couple of lines below.
Not completely sure if it would be helpful, but, maybe, we can add readability to the code.
Mostly because the method names are quite similar and looking, for the first time, through the file I got lost.

Reply:
Yeah, I don't like these methods, I would like them to be improved too.